### ISSUE TITLE
assertIsNotNone(foo) in tests is not treated the same as `assert foo is not None`

### ISSUE THREAD

**Issue by timabbott** [python/mypy#5088]
As part of Zulip's strict-optional effort, I had to convert a test that was using `assertIsNotNone(foo)` to use `assert foo is not None` in order to get mypy to pass.  I think mypy should treat those two the same.

**Comment by JelleZijlstra** [comment]
Is `assertIsNotNone` here a custom function? One way to implement this could be to type `assertisNotNone` as:
```python
@overload
def assertisNotNone(obj: None) -> NoReturn: ...
@overload
def assertIsNotNone(obj: Any) -> None: ...
```
Haven't tested it though.

**Comment by elazarg** [comment]
@JelleZijlstra that's #4063. It is not implemented.

**Comment by timabbott** [comment]
It's part of the standard unittest library: https://docs.python.org/2/library/unittest.html#unittest.TestCase.debug

**Comment by ethanhs** [comment]
I think this can be closed in favor of #4063 then.

**Comment by 1kastner** [comment]
> Is `assertIsNotNone` here a custom function? One way to implement this could be to type `assertisNotNone` as:
> 
> ```python
> @overload
> def assertisNotNone(obj: None) -> NoReturn: ...
> @overload
> def assertIsNotNone(obj: Any) -> None: ...
> ```
> 
> Haven't tested it though.

Is that something that should be added to the typeshed description of unittest then? I checked https://github.com/python/typeshed/blob/master/stdlib/unittest/case.pyi#L120 which reads this:

> def assertIsNotNone(self, obj: object, msg: Any = ...) -> None: ...

I have seen the forward to https://github.com/python/mypy/issues/4063. However, that issue is open for 5 years with no result. And currently I prefer to have this issue solved. I just do not want to double-check everything with both an assert statement and a unittest check. I also do not want to refactor all unittests of a project just because of a missing functionality of mypy. I hope that makes it understandable why it might be more reasonable to pick up this track. Thank you!
